WordPress/gutenberg
# Pages: Add "Set as posts page" action

## What?

https://github.com/WordPress/gutenberg/pull/65426 added the Set as Homepage action to Pages, so this is a follow-up which adds a "Set as posts page" action as well.

It also includes a few other smaller tweaks:

- Ensures the "Move to trash" action is always at the bottom of the actions list
- Moves `getItemTitle` to its own file as it's now being used in two actions
- Removes the use of `saveEditedEntityRecord` in the "Set as homepage" and "Set as posts page" actions so it's more consistent with other similar use cases

## Why?

Addresses part of the second task in [this comment](https://github.com/WordPress/gutenberg/pull/65426#issuecomment-2506012109).

This means that more of the homepage display controls from Settings -> Reading are replicated within the Editor, which hopefully makes changing them easier from within the Editor.

## How?

By adding one new action, "Set as posts page".

## Testing Instructions




1. Ensure you have not yet set a homepage or a posts page
2. Go to Editor -> Pages
3. Select the Actions menu
4. Select 'Set as posts page'
5. Ensure the selected page is set as the posts page

## Screenshots or screencast 



Both "set as" actions:
<img width="402" alt="image" src="https://github.com/user-attachments/assets/4d01d63a-9c19-422a-b42c-70426a744f57" />

Posts page confirm modal:
<img width="537" alt="image" src="https://github.com/user-attachments/assets/f47d28ca-0868-4552-8d3f-9239caae0019" />
